README.md

```diff
@@ -3,33 +3,71 @@
 __URLs__
 
 * Name all of the parts of the url that you can remember.  In your own words describe what they do.
+
+```Protocol - either http or https. Just establishes which protocol is being used.
+```
+```Domain - a name that directs to the correct server address.
+```
+```Port - a specific program reference.
+```
+```Path - the specific folder location being directed to.
+```
+```Query - (optional) if there is some input on the page that needs to be duplicated. Such as a search.
+```
+```Link - a specific link on the HTML to direct to.
+```
+
 * Name the pieces of the following urls:
-	* `https://www.google.com/`
-	* `https://workbook.galvanize.com/cohorts/41/learning_experiences/367`
-	* `http://locahost:5000/animals/puppies?onlycute=1&size=medium#firstpuppy`
-	* `https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error`
+	* `https://www.google.com/` ```protocol/domain```
+	* `https://workbook.galvanize.com/cohorts/41/learning_experiences/367` ```protocol/domain/path```
+	* `http://locahost:5000/animals/puppies?onlycute=1&size=medium#firstpuppy` ```protocol/domain/port/path/query/link```
+	* `https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error` ```protocol/domain/path/link```
 * Can a server use more than 1 port?
+```I don't believe so
+```
 * Why is https different than http?
-* How does a server interpret the following url's query paramter.  What data structure does it create on the server?
+```Https uses a secured connection
+```
+* How does a server interpret the following url's query parameter.  What data structure does it create on the server?
 
 ```
 http://locahost:5000/animals?puppies=fido&puppies=max&puppies=moxie
 ```
 
+```It will only pull up the puppies, Fido, Max, and Moxie.
+```
+
 __HTTP Request/Response__
 
 * Name at least 4 http verbs
+```GET, POST, PUT, DELETE
+```
 * What is each verb useful for in your own words
+```GET pulls down the requested information, usually the html to be rendered.
+```POST allows new data to be uploaded.
+```PUT allows for existing data to be modified.
+```and DELETE removes data.
+```
 * What does idempotent mean?
+```Looking at a resource without changing it.
+```
 * Name the 5 http status code ranges.  What are they used for in general?
+```500: an error
+```400: a resource that doesn't exist, no error server side.
```

Comment:
user error: asking for a resource that doesn't exist, sending xml instead of json, sending all the wrong query params etc..